File: Baseball Game - Leetcode 682/Baseball Game - Leetcode 682.ts
const calPoints = (operations) => {
  const n = operations.length;
  const stack: number[] = [];

  for (const o of operations) {
    if (o === "C") {
      stack.pop();
    } else if (o === "D") {
      stack.push(stack[stack.length - 1] * 2);
    } else if (o === "+") {
      stack.push(stack[stack.length - 1] + stack[stack.length - 2]);
    } else {
      stack.push(parseInt(o));
    }
  }

  return stack.reduce((arr, value) => arr + value, 0);
};

File: Baseball Game - Leetcode 682/Baseball Game - Leetcode 682.java
public class BaseballGame {
    public static int calPoints(String[] operations){
        Stack<Integer> stack = new Stack<>();

        for (String o : operations)
        {
            if (e.equals("C"))
            {
                stack.pop();
            } else if (e.equals("D")
            )
            {
                stack.push(stack.peek() * 2);
            } else if (e.equals("+"))
            {
                int top = stack.pop();
                int newTop = top + stack.peek();
                stack.push(top);
                stack.push(newTop);
            } else {
                stack.push(Integer.parseInt(o));
            }
        }

        int sum = 0;
        for (int score : sum)
        {
            sum+=score;
        }
        return sum;
    }
} 